examples/sqlite/main.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+import os
+import sqlite3
+import pathlib
+
+from typing import Dict
+
+from nicegui import ui
+
+# initialize sqlite file
+DB_FILE = os.path.join(os.path.dirname(__file__), "db", "people.db")
+pathlib.Path(DB_FILE).touch()
+
+conn = sqlite3.connect(DB_FILE, check_same_thread=False)
+cursor = conn.cursor()
+cursor.execute(
+    "CREATE TABLE IF NOT EXISTS users (id integer primary key AUTOINCREMENT, name text, age integer)"
+)
+conn.commit
+
+# UI starts here
+name = ui.input(label="username").classes("w-full")
+age = ui.number(label="your age").classes("w-full")
+
+get_id = ui.label()
+
+list_alldata = ui.column()
+
+with ui.dialog() as editdialog:
+    with ui.card():
+        ui.label("Edit Data").classes("font-xl font-weight")
+        edit_name = ui.input().bind_value_from(name, "value")
+        edit_age = ui.input().bind_value_from(age, "value")
+
+        with ui.row().classes("justify-between"):
+            ui.button("save", on_click=lambda e: save_edit(e))
+            ui.button("close", on_click=editdialog.close).classes("bg-red")
+
+
+def add_newdata():
+    """Callback when a new user button is clicked"""
+    try:
+        cursor.execute(
+            "INSERT INTO users (name, age) VALUES (?, ?)", (name.value, age.value)
+        )
+        conn.commit()
+        ui.notify(f"success add new user {name.value}", color="blue")
+        name.value = ""
+        age.value = ""
+        list_alldata.clear()
```

Comment:
You could take advantage of [ui.refreshable](https://nicegui.io/documentation#refreshable_ui_functions) to simplify the code.